IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -146,4 +150,16 @@ + (NSString *)createDeviceAuthResponse:(NSString *)audience
     return [MSIDJWTHelper createSignedJWTforHeader:header payload:payload signingKey:[identity privateKeyRef]];
 }
 
++(void)saveTelemetryForAdfsPkeyAuthChallengeForUrl:(NSURL *)submitUrl
+                                              code:(NSUInteger)code
+                                           context:(id<MSIDRequestContext>)context
+{
+    NSString *submitUrlString = submitUrl.absoluteString;
+    if ([submitUrlString containsString:@"/adfs/ls/"] || [submitUrlString containsString:@"/adfs/oauth2/"])
+    {
+        MSIDAADTokenRequestServerTelemetry *serverTelemetry = [MSIDAADTokenRequestServerTelemetry new];
+        NSString *telemetryMessage = [NSString stringWithFormat:@"%@_%@",@"ADFS_PKEYAUTH_CHLG",submitUrl.host];
+        [serverTelemetry handleError:[[NSError alloc] initWithDomain:telemetryMessage code:code userInfo:nil] context:context];
```

Comment:
Does this go into the next last telemetry request?

Reply:
Yes, it is sent in x-client-last-telemetry header and shows up in MsalLast column of ESTS kusto

Reply:
Thanks, looking good!